pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -506,15 +509,49 @@ def cached_async_http_client(*, provider: str | None = None, timeout: int = 600,
 @cache
 def _cached_async_http_client(provider: str | None, timeout: int = 600, connect: int = 5) -> httpx.AsyncClient:
     return httpx.AsyncClient(
-        transport=_cached_async_http_transport(),
+        transport=_get_transport_for_loop(),
         timeout=httpx.Timeout(timeout=timeout, connect=connect),
         headers={'User-Agent': get_user_agent()},
     )
 
 
 @cache
-def _cached_async_http_transport() -> httpx.AsyncHTTPTransport:
-    return httpx.AsyncHTTPTransport()
+def _get_transport_for_loop() -> _PerLoopTransport:
+    return _PerLoopTransport()
+
+
+# transport._pool._connections[0]._connection._network_stream._stream.transport_stream._transport._loop
+class _PerLoopTransport(httpx.AsyncBaseTransport):
+    def __init__(self):
+        self.transports: MutableMapping[asyncio.AbstractEventLoop, httpx.AsyncHTTPTransport] = (
+            weakref.WeakKeyDictionary()
+        )
+
+    def get_transport(self) -> httpx.AsyncHTTPTransport:
+        # Clean the dictionary of closed loops
+        for loop in list(self.transports.keys()):
+            if loop.is_closed():
+                del self.transports[loop]
+
+        return self.transports.setdefault(asyncio.get_running_loop(), httpx.AsyncHTTPTransport())
+
+    async def __aenter__(self):
+        await self.get_transport().__aenter__()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None = None,
+        exc_value: BaseException | None = None,
+        traceback: TracebackType | None = None,
+    ) -> None:
+        await self.get_transport().__aexit__(exc_type, exc_value, traceback)
```

Comment:
Those seem to be the not-covered lines. Meaning we (or rather httpx) never use the aenter/aexit API of the transport. We could just raise NotImplementError and rather that do something not obvious and not needed wait for someone to actually need it ?

I'm saying "not obvious" because it's very slightly unclear whether we would want `__aenter__` to return `self` or the underlying transport